pkg/api/v20191231preview/openshiftcluster.go

```diff
@@ -79,6 +81,13 @@ const (
 	ProvisioningStateFailed        ProvisioningState = "Failed"
 )
 
+func ToExternalProvisioningState(state api.ProvisioningState) ProvisioningState {
+	if state == "OperatorUpdating" {
+		return ProvisioningStateAdminUpdating
+	}
+	return ProvisioningState(state)
+}
```

Comment:
I somehow feel that this is a bad idea to have 1 external provisioning state for >=2 internal states.

Reply:
Also how do we put the document into `OperatorUpdating` state? Sounds like we need to modify [`putOrPatchOpenShiftCluster`](https://github.com/Azure/ARO-RP/blob/f9eaa7ffe8b6c431325d71e6e7532092ac605d21/pkg/frontend/openshiftcluster_putorpatch.go#L191-L198) somehow (introduce another special case?).

Reply:
What are the alternative ways to implement this?

Maybe we can re-use `AdminUpdating` and parametrise it (e.g. update all, update operator only, etc)?

Reply:
+1 on re-use the state

Reply:
Done on the new version.